tcping.go

```diff
@@ -671,15 +685,22 @@ func (tcpStats *stats) handleConnSuccess(rtt float32, connTime time.Time, elapse
 
 // tcping pings a host, TCP style
 func tcping(tcpStats *stats) {
-	IPAndPort := netip.AddrPortFrom(tcpStats.userInput.ip, tcpStats.userInput.port)
-
+	var err error
+	var conn net.Conn
 	connStart := time.Now()
-	// if timeout = 0, there is no timeout, see net.Dial
-	conn, err := net.DialTimeout("tcp", IPAndPort.String(), tcpStats.userInput.timeout)
+
+	if tcpStats.userInput.networkInterface.use {
+		conn, err = tcpStats.userInput.networkInterface.dailer.Dial("tcp", tcpStats.userInput.networkInterface.raddr.String())
```

Comment:
I think it is better to explicitly use the [DialTimeout ](https://cs.opensource.google/go/go/+/refs/tags/go1.21.0:src/net/dial.go;l=413) instead of `Dial`. This way, it'll be easier for others to read the code and contribute. At the first glance without digging through the `newNetworkInterface` function, one might think there is a mistake.

Reply:
problem is I can't pass `local address` or `interface` to the `net.DialTimeout()` as it creates its own `Dailer` with default values..
I can add some comments.

Reply:
ah dang it.